metatomic-torch/include/metatomic/torch/io/io.hpp

```diff
@@ -0,0 +1,34 @@
+#pragma once
+
+#include <cstddef>
+#include <cstdint>
+#include <string>
+#include <vector>
+#include <torch/torch.h>
+
+namespace metatomic_torch {
+
+// From system.hpp
+class NeighborListOptionsHolder;
+using NeighborListOptions = torch::intrusive_ptr<NeighborListOptionsHolder>;
+
+class SystemHolder;
+using System = torch::intrusive_ptr<SystemHolder>;
+
+// ===== File-based =====
+void   save_system_file(const std::string& mta_path, const System& system);
+System load_system_file(const std::string& mta_path);
+bool   is_system_mta_file(const std::string& mta_path);
+
+// ===== In-memory (BytesIO-like) =====
+std::vector<uint8_t> save_system_memory(const System& system);
+System               load_system_memory(const uint8_t* data, size_t size);
+inline System        load_system_memory(const std::vector<uint8_t>& buf) {
+  return load_system_memory(buf.data(), buf.size());
+}
```

Comment:
Following the same naming convention as metatensor

```suggestion
std::vector<uint8_t> save_buffer(const System& system);
System               load_system_buffer(const uint8_t* data, size_t size);
inline System        load_system_buffer(const std::vector<uint8_t>& buf) {
  return load_system_buffer(buf.data(), buf.size());
}
```